### ISSUE TITLE
RFE: Shortcut for recent archives

### ISSUE THREAD

**Issue by myllynen** [performancecopilot/pcp#408]
Consider a typical use case where the admin wants to check some recent activity on a system with sar(1):

```
$ sar -w
```

This automatically fetches the requested information for current day (from sar archives under /var/log/sa or wherever).

It would be nice to have such a default or shortcut(s) for PCP client tools as well in libpcp for more pleasant end-user experience, for example

```
$ pmval -a TODAY mem.util.used
$ pmrep -a LOCAL :sar-w
```

would automagically use the local archives from the default location (so that the used wouldn't have to guess the directory nor the names of the relevant archives) and the client tools would then print the values just like the case of the user specifying the archive(s) in question using the current way of ... -a /var/log/pcp/pmlogger/fedora-27-server/20180103.15.10 ... (which won't work tomorrow even on the current system).

Thanks.

**Comment by goodwinos** [comment]
It's a good idea, and one I've often though about but more along the lines of having a default archive $PATH or some such. e.g. $PCP_LOG_DIR/pmlogger/$(hostname) and rely on multi-archive to scan the directory (or the equivalent pmmgr directories).

With a default archive PATH, the TODAY concept could perhaps be a default -S or -O start time selection / synonym for replay from the default archive path. i.e. -S TODAY.

Another issue is having a default archive PATH confuses defaulting to -h local (live mode if neither -h or -a are given), so we'd still need -a SOMETHING to request archive replay rather than live mode.

**Comment by natoscott** [comment]
+1 - great idea.

I think overloading -a/--archive is not the way to go, though, I'd prefer to just use the presence of -O to distinguish where in the archive(s) to start from, and the fact that archives should be used over live.  

The common libpcp getopt-handling code could then contain the find-the-default-path heuristics, and automatically insert that into the pmOptions struct.

**Comment by goodwinos** [comment]
OK, so -O TODAY would also implicitly specify -a $PCP_LOG_DIR/pmlogger/$(hostname)
(any of the time-window options would imply archive mode).

**Comment by natoscott** [comment]
@goodwinos yep, and the "-O today handling code is already in libpcp ...
grep today ../../src/libpcp/src/getdate.y

**Comment by goodwinos** [comment]
@natoscott how are we going with the archive/local unification? - it's on the pcp-4.0.1 roadmap.
Would be handy to also implement pmFetchWait() to block if the archive position fetch time is in the near future, so we could say pmtool -Onow ... and automatically logtail from the default archive in "almost-live" mode.

**Comment by natoscott** [comment]
@goodwinos no progress to report there unfortunately, too many distractions recently.  A log-tail-following mode for archive contexts would be very handy though (as per your notes from awhile back), and would resolve the PMAPI side of things well enough I think (a separate issue to the pmSeries API side of things described on the roadmap - which needs that as well).

**Comment by natoscott** [comment]
commit e6fe52c4b2eb893384c6e59780e8e6f2916c2b9e
Author: Nathan Scott <nathans@redhat.com>
Date:   Mon Jan 8 11:45:08 2018 +1100

    libpcp: provide -O/--offset heuristics for local archives
    
    Implement a shorthand notation for the -O/--origin option,
    allowing the local system archive directory to be inserted
    as an archive name when no other source of metrics has been
    supplied.
    
    This was complicated by a quirk of the libpcp getdate.y code
    where "today" is (was) an exact synonym for "now".  In this
    new mode of operation, we want "today" to mean "the start of
    today", instead (& similarly with semantics of "yesterday").
    This behaviour been changed in libpcp since its likely to be
    a desirable interpreation of these terms in all PCP cases.
    
    New qa test 625 exercises this new -O option behaviour, and
    the PCPIntro(1) man page documents it as well.
    
    Resolves github issue #408.

**Comment by fche** [comment]
> [...] Would be handy to also implement pmFetchWait()  [...]

Maybe this part isn't needed much.  The same pmapi client code that would know enough that it has to call pmFetchWait(X) vs. pmFetch() could already do a sleep(X); pmFetch() sequence.

See also https://github.com/performancecopilot/pcp/issues/178 .
